src/quo2/components/icon.cljs

```diff
@@ -12,33 +12,37 @@
       (and (string? color)
            (not (string/blank? color)))))
 
+(defn- image-icon-style
+  [{:keys [color no-color size container-style theme]}]
+  (cond-> {:width  size
+           :height size}
+    (not no-color)
+    (assoc :tint-color
+           (if (and (string? color) (not (string/blank? color)))
+             color
+             (colors/theme-colors colors/neutral-100 colors/white theme)))
+    :always
+    (merge container-style)))
+
 (defn memo-icon-fn
-  [{:keys [color color-2 no-color
-           container-style size accessibility-label theme]
-    :or   {accessibility-label :icon}}
+  [{:keys [color color-2 container-style size accessibility-label]
+    :or   {accessibility-label :icon}
+    :as   props}
    icon-name]
   (let [size (or size 20)]
-    ^{:key icon-name}
-    (if-let [svg-icon (icons.svg/get-icon icon-name size)]
-      [svg-icon
-       {:size                size
-        :color               (when (valid-color? color) color)
-        :color-2             (when (valid-color? color-2) color-2)
-        :accessibility-label accessibility-label
-        :style               container-style}]
-      [rn/image
-       {:style
```

Comment:
This metadata is not being added properly. I fixed it and also refactored the code a little. To show what's wrong with the metadata:

```clojure
;; Before
(meta (let [_ nil]
        ^{:key "HOLA"} ;; <- here
        (if-let [_ true]
          ['my-component 'params]
          ['my-component-2 'params-2])))
;; => nil

;; Now
(meta (let [_ nil]
        (with-meta ;; <- here
         (if-let [_ true]
           ['my-component 'params]
           ['my-component-2 'params-2])
         {:key "HOLA"})))
;; => {:key "HOLA"}

```